File: 서예진A/20230517/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int a=0;a<commands.size();a++){
        int i=commands[a][0];
        int j=commands[a][1];
        int k=commands[a][2];
        
        vector<int> arr; //자른 배열 저장 vector
        
        for(int l=i-1;l<j;l++){ //i~j번째까지 배열 자르기
            arr.push_back(array[l]);
        }
        
        sort(arr.begin(),arr.end()); //arr 정렬
        
        answer.push_back(arr[k-1]); //연산 적용 시 나온 결과 배열에 넣기
        
    }
    return answer;
}

File: 서예진B/20230517/K번째수.py
def solution(array, commands):
    answer = []
    for command in commands:
        i, j, k = command
        # i번째 ~ j번째 수를 담은 배열 num (인덱싱 : array[시작인덱스:끝인덱스+1])
        num = array[i-1:j]
        num.sort()
        # k번째 수를 answer 배열에 append
        answer.append(num[k-1])
    return answer